### DISCUSSION TITLE
import  html editor question

### DISCUSSION THREAD

**Discussion by Hsin90321** [GrapesJS/grapesjs#4159] [Q&A]
Hi , I use this comment to editor my html code
https://github.com/artf/grapesjs/issues/324#issuecomment-330571539

import & editor are perfect,But today I have question.
if I use custom data-gjs-type in html , and import again
data-gjs-type='xxxx' will become 'defalut'
what should I do?

↓ init tabs componets, tab is great
![1645770550125](https://user-images.githubusercontent.com/55831138/155667406-671a7a61-d78f-4275-995b-197dca446055.jpg)

↓ open html editor and edit text , and click import button
![1645770575532](https://user-images.githubusercontent.com/55831138/155667455-f3070084-96bf-4eac-b425-e7696c032770.jpg)

↓ data-gjs-type="tabs"  become 'default' , tab is not working 
![1645770598012](https://user-images.githubusercontent.com/55831138/155667463-0cf28276-3a9f-42d2-a522-f5a0d2954f33.jpg)

**Comment by artf** [comment]
The exported HTML is optimized and doesn't contain enough information for the editor to understand the code (eg. `data-gjs-type` attribute).
One thing you can do is to make use of `withProps` option (available from https://github.com/artf/grapesjs/releases/tag/v0.18.2) in [`editor.toHTML`](https://grapesjs.com/docs/api/component.html#tohtml).
You can also enforce this option on init
```js
grapesjs.init({
  // ...
  optsHtml: { withProps: true }
})
```
